docs/src/main/sphinx/connector/sqlserver.rst

```diff
@@ -119,26 +119,186 @@ that catalog name instead of ``sqlserver`` in the above examples.
-``tinyint``                         ``smallint``
-``smallint``                        ``smallint``
-``int``                             ``integer``
-``float``                           ``double``
-``char(n)``                         ``char(n)``
-``varchar(n)``                      ``varchar(n)``
-``date``                            ``date``
-``datetime2(n)``                    ``timestamp(n)``
-``datetimeoffset(n)``               ``timestamp(n) with time zone``
-==================================  ===============================
+Because Trino and SQL Server each support types that the other does not, this
+connector modifies some types when reading or writing data.
+
+SQL Server to Trino read type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This connector supports reading the following SQL Server types and performs
+conversion to Trino types with the detailed mappings as shown in the following
+table.
+
+.. list-table:: SQL Server to Trino type mapping
+  :widths: 30, 20, 50
+  :header-rows: 1
+
+  * - SQL Server database type
+    - Trino type
+    - Notes
+  * - ``BIT``
+    - ``BOOLEAN``
+    -
+  * - ``TINYINT``
+    - ``SMALLINT``
+    - SQL Server ``TINYINT`` is actually ``unsigned tinyint``
+  * - ``SMALLINT``
+    - ``SMALLINT``
+    -
+  * - ``INTEGER``
+    - ``INTEGER``
+    -
+  * - ``BIGINT``
+    - ``BIGINT``
+    -
+  * - ``DOUBLE PRECISION``
+    - ``DOUBLE``
+    -
+  * - ``FLOAT[(n)]``
+    - ``REAL`` or ``DOUBLE``
+    -  See :ref:`sqlserver-numeric-mapping`
+  * - ``REAL``
+    - ``REAL``
+    -
+  * - ``MONEY``
+    - ``DECIMAL(19, 4)``
+    -
+  * - ``SMALLMONEY``
+    - ``DECIMAL(10, 4)``
+    -
+  * - ``DECIMAL[(p[, s])]``, ``NUMERIC[(p[, s])]``
+    - ``DECIMAL(p, s)``
+    -
+  * - ``CHAR[(n)]``
+    - ``CHAR(n)``
+    - ``1 <= n <= 8000``
+  * - ``NCHAR[(n)]``
+    - ``CHAR(n)``
+    - ``1 <= n <= 4000``
+  * - ``VARCHAR[(n | max)]``, ``NVARCHAR[(n | max)]``
+    - ``VARCHAR(n)``
+    - ``1 <= n <= 8000``, ``max = 2147483647``
+  * - ``TEXT``
+    - ``VARCHAR(2147483647)``
+    -
+  * - ``NTEXT``
+    - ``VARCHAR(1073741823)``
+    -
+  * - ``uniqueidentifier``
+    - ``VARCHAR(36)``
+    -
+  * - ``XML``
+    - ``VARCHAR(2147483647)``
+    -
+  * - ``BINARY``, ``VARBINARY``, ``IMAGE``, ``TIMESTAMP``, ``UDT``, ``GEOMETRY``, ``GEOGRAPHY``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``DATE``
+    -
+  * - ``TIME[(n)]``
+    - ``TIME(n)``
+    - ``0 <= n <= 7``
+  * - ``DATETIME``
+    - ``TIMESTAMP(3)``
+    -
+  * - ``DATETIME2[(n)]``
+    - ``TIMESTAMP(n)``
+    - ``0 <= n <= 7``
+  * - ``SMALLDATETIME``
+    - ``TIMESTAMP(0)``
+    -
+  * - ``DATETIMEOFFSET[(n)]``
+    - ``TIMESTAMP(n) WITH TIME ZONE``
+    - ``0 <= n <= 7``
+
+Trino to SQL Server write type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This connector supports writing the following Trino types and performs
+conversion to SQL Server types with the detailed mappings as shown in the
+following table.
```

Comment:
See comment above. Would suggest rewording to "The connector maps Trino to the corresponding SQL Server types following this table:"